### PR TITLE
Add rustc_codegen_gcc repo under automation

### PR DESCRIPTION
**Pull request by Kobzol**
Repo: https://github.com/rust-lang/rustc_codegen_gcc

CC @antoyo

Extracted from GH:
```
org = "rust-lang"
name = "rustc_codegen_gcc"
description = "libgccjit AOT codegen for rustc"
bots = []

[access.teams]
security = "pull"
compiler = "maintain"

[access.individuals]
estebank = "maintain"
Aaron1011 = "maintain"
pnkfelix = "maintain"
eddyb = "maintain"
petrochenkov = "maintain"
matthewjasper = "maintain"
cjgillot = "maintain"
pietroalbini = "admin"
jackh726 = "maintain"
rust-lang-owner = "admin"
wesleywiser = "maintain"
rylev = "admin"
davidtwco = "maintain"
michaelwoerister = "maintain"
oli-obk = "maintain"
lcnr = "maintain"
compiler-errors = "maintain"
nagisa = "maintain"
jdno = "admin"
antoyo = "write"
Mark-Simulacrum = "admin"
badboy = "admin"
```

### CODE REVIEW

File: repos/rust-lang/rustc_codegen_gcc.toml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+org = "rust-lang"
+name = "rustc_codegen_gcc"
+description = "libgccjit AOT codegen for rustc"
+bots = []
+
+[access.teams]
+compiler = "maintain"
+
+[access.individuals]
+antoyo = "maintain"
```
Comment (GuillaumeGomez) [inline repos/rust-lang/rustc_codegen_gcc.toml:10]: Would be nice to list me here as well. :)

**Comment by GuillaumeGomez** [comment]
What exactly does it add?

**Comment by Kobzol** [comment]
It doesn't really add anything, I just want to move the repo under automation. Currently, most of our repos are managed manually, and I'm now doing a sweep across these and moving them under `team` one by one :)

**Comment by GuillaumeGomez** [comment]
Nice! Glad to see more automation coming in. :)

File: repos/rust-lang/rustc_codegen_gcc.toml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+GuillaumeGomez = "maintain"
```
Comment (rylev) [inline repos/rust-lang/rustc_codegen_gcc.toml:11]: Does @GuillaumeGomez need maintain capabilities? I don't believe the have even write permission currently so this is a pretty massive bump in privileges. cc @antoyo

Comment (antoyo) [reply repos/rust-lang/rustc_codegen_gcc.toml:11]: I'm not even sure what that even means. Is this to manage write access to the git repositories? If so, I believe @GuillaumeGomez doesn't currently have write access indeed.

Comment (antoyo) [reply repos/rust-lang/rustc_codegen_gcc.toml:11]: To clarify, I'm good with giving @GuillaumeGomez write access to the repo.

Comment (Kobzol) [reply repos/rust-lang/rustc_codegen_gcc.toml]: The permissions are described [here](https://docs.github.com/en/organizations/managing-user-access-to-your-organizations-repositories/managing-repository-roles/repository-roles-for-an-organization.

TLDR:
- write => can push to branches and merge PRs
- maintain => can also manage wikis, push to protected branches, and a bunch of other stuff.

Actually, I should have arguably used `write` here. I kept `maintain` only for you now.

Comment (GuillaumeGomez) [reply repos/rust-lang/rustc_codegen_gcc.toml]: Then `write` is better for me. I'll leave the rest to @antoyo. :3

**Review by antoyo — APPROVED** [review]

**Review by rylev — APPROVED** [review]
